src/jit/lsra.cpp

```diff
@@ -5678,10 +5669,20 @@ void LinearScan::allocateRegisters()
                     allocateReg = false;
                 }
 
+#if FEATURE_PARTIAL_SIMD_CALLEE_SAVE && defined(_TARGET_XARCH_)
+                // We can also avoid allocating a register (in fact we don't want to) if we have
+                // an UpperVectorRestore on xarch where the value is on the stack.
+                if ((currentRefPosition->refType == RefTypeUpperVectorRestore) && (currentInterval->physReg == REG_NA))
+                {
+                    assert(currentRefPosition->regOptional);
+                    allocateReg = false;
+                }
+#endif
+
 #ifdef DEBUG
                 // Under stress mode, don't attempt to allocate a reg to
-                // reg optional ref position.
-                if (allocateReg && regOptionalNoAlloc())
+                // reg optional ref position, unless it's a ParamDef.
+                if (allocateReg && regOptionalNoAlloc() && (currentRefPosition->refType != RefTypeParamDef))
```

Comment:
I added the second condition here because, previously, we never marked these as `RegOptional`. In cleaning up the weird semantics of `RequiresRegister`, this behavior under stress was changed (specifically jitStressRegs 0x1000, which is the `regOptionalNoAlloc`). `RefTypeParamDef`s aren't considered "ActualRefs" but previously weren't marked `RegOptional`. However, the x86 gc encoding doesn't seem to expect the 'this' ptr not to get a register (it can be spilled, but there's a dependency somewhere on it getting an initial register assignment). I haven't debugged this thoroughly, but will open an issue after this PR is merged to investigate why it's not handled properly. As it is, this reverts the behavior for these `RefPosition`s to what it used to be.